### PR TITLE
fix: item with colon in map can not be parsed correctly

### PR DESCRIPTION
**Pull request by Yiyiyimu**
Signed-off-by: yiyiyimu <wosoyoung@gmail.com>

fix https://github.com/apache/apisix/issues/4449

Inline nested hash should have an extra space after the colon

https://github.com/peposso/lua-tinyyaml/blob/e11d16d21200846719a8e03b7f0d00ff37c5b961/tinyyaml.lua#L513-L514

---

Also add CI for easier test and test passed in my forked repo
https://github.com/Yiyiyimu/lua-tinyyaml/actions/runs/953567969

### CODE REVIEW

**Comment by Yiyiyimu** [comment]
Hi @spacewander we need to merge upstream first

**Comment by spacewander** [comment]
We should use a separate PR to sync the upstream.
Also, need a way to work around the CLA...

**Comment by Yiyiyimu** [comment]
> We should use a separate PR to sync the upstream.

Got it. Fixed by #3

**Comment by Yiyiyimu** [comment]
Rebased
PTAL @spacewander

File: tinyyaml.lua [hunk 1]
```diff
@@ -510,7 +510,7 @@ local function parseseq(line, lines, indent)
     end
     local rest = ssub(line, j+1)
 
-    if sfind(rest, '^[^\'\"%s]*:') then
+    if sfind(rest, '^[^\'\"%s]*:%s') then
```
Comment (spacewander) [inline tinyyaml.lua:513]: Is there RFC or defacto implementation requires the hash value to start with whitespace?

Comment (Yiyiyimu) [reply tinyyaml.lua:513]: Failed to go through official yaml spec, but tested on https://onlineyamltools.com/convert-yaml-to-json
image
image

**Review by spacewander — APPROVED** [review]

**Pull request merged**